Xamarin.Essentials/DeviceDisplay/DeviceDisplay.shared.cs

```diff
@@ -6,8 +6,16 @@ public static partial class DeviceDisplay
     {
         static event EventHandler<ScreenMetricsChangedEventArgs> ScreenMetricsChangedInternal;
 
+        static ScreenMetrics currentMetrics;
+
         public static ScreenMetrics ScreenMetrics => GetScreenMetrics();
 
+        static void SetCurrent()
+        {
+            var metrics = GetScreenMetrics();
+            currentMetrics = new ScreenMetrics(metrics.Width, metrics.Height, metrics.Density, metrics.Orientation, metrics.Rotation);
```

Comment:
~Why do we copy the struct, can we not assign it directly because it will make a copy anyway?~ ah, I see we have a readonly struct.